generators/entity-server/templates/src/main/java/package/common/patch_template.ejs

```diff
@@ -0,0 +1,56 @@
+<%#
+ Copyright 2013-2020 the original author or authors from the JHipster project.
+
+ This file is part of the JHipster project, see https://www.jhipster.tech/
+ for more information.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+-%>
+<%
+const instanceType = (dto === 'mapstruct') ? asDto(entityClass) : asEntity(entityClass);
+const instanceName = (dto === 'mapstruct') ? asDto(entityInstance) : asEntity(entityInstance);
+const returnType = reactive ? 'Mono' : 'Optional'
+const mapOrFlatMap = reactive ? 'flatMap' : 'map'
+const returnPrefix = (isService) ? 'return' : returnType + '<' + instanceType + '> result =';
+const mapper = entityInstance  + 'Mapper';
+_%>
+<%_ if(viaService) { %>
+   <%_ if(!reactive) { %>
+   Optional<<%= instanceType %>> result = Optional.ofNullable(<%= entityInstance %>Service.partialUpdate(<%= instanceName %>));
+   <%_ } else { %>
+   Mono<<%= instanceType %>> result = <%= entityInstance %>Service.partialUpdate(<%= instanceName %>);
+   <%_ } %>
+<%_ } else { %>
+<%- returnPrefix %> <%= entityInstance %>Repository.findById(<%= instanceName %>.getId())
+   .map(existing<%= entityClass %> -> {
+<%_ for (field of fields) { _%>
```

Comment:
Since we are doing patch updates, maybe should randomize fields to include:
```
const includeField = faker.random.boolean()
if (includeField) { %> /* Randomized disable for patch test <%};
...
if (includeField) { %> */<%};
```

Reply:
And is possible to add new fields, instead of just set the old value.
https://github.com/jhipster/generator-jhipster/blob/05e4ab9cec320c4e22260ae8a6d99f13aea4b315/generators/entity-client/templates/common/src/test/javascript/cypress/integration/entity/entity.spec.ts.ejs#L29
https://github.com/jhipster/generator-jhipster/blob/05e4ab9cec320c4e22260ae8a6d99f13aea4b315/generators/entity-client/templates/common/src/test/javascript/cypress/integration/entity/entity.spec.ts.ejs#L146

Reply:
> Since we are doing patch updates, maybe should randomize fields to include:
> 
> ```
> const includeField = faker.random.boolean()
> if (includeField) { %> /* Randomized disable for patch test <%};
> ...
> if (includeField) { %> */<%};
> ```

Can you give me more details please? I don't really understand what you mean

Concerning your second comment, we didn't generate any view for the patch, I don't understand why you're quoting cypress tests, can you explain please?

Reply:
Let's keep this way.